src/VisualStudio/Core/Test/CodeModel/CSharp/RootCodeModelTests.vb

```diff
@@ -50,6 +50,23 @@ namespace N
         End Function
 
         <ConditionalWpfFact(GetType(x86)), Trait(Traits.Feature, Traits.Features.CodeModel)>
+        Public Async Function TestDotNetNameFromLanguageSpecific2() As Task
+            Await TestRootCodeModelWithCodeFile(<code></code>,
+                Sub(rootCodeModel)
+                    Dim dotNetName = rootCodeModel.DotNetNameFromLanguageSpecific("System.Collections.Generic.List<int>")
+                    Assert.Equal("System.Collections.Generic.List`1[System.Int32]", dotNetName)
+                End Sub)
+        End Function
+
+        <ConditionalWpfFact(GetType(x86)), Trait(Traits.Feature, Traits.Features.CodeModel)>
+        Public Async Function TestDotNetNameFromLanguageSpecificWithAssemblyQualifiedName() As Task
+            Await TestRootCodeModelWithCodeFile(<code></code>,
+                Sub(rootCodeModel)
+                    Assert.Throws(Of ArgumentException)(Sub() rootCodeModel.DotNetNameFromLanguageSpecific("System.Collections.Generic.List<int>, mscorlib"))
```

Comment:
Just for completeness, it might be worth using a name with a comma in the type parameter list, as well, since that might otherwise be valid.